### ISSUE TITLE
ITU Y.3507 document review (continued from #60)

### ISSUE THREAD

**Issue by anermina** [metanorma/mn-native-pdf#158]
In https://github.com/metanorma/mn-native-pdf/issues/60#issuecomment-582468325, it is written that some fixes were made for this document, but the issues are unfortunately still present in PDF uploaded 2 days ago. Could you please try generating PDF again?

> > Title is not rendering properly in PDF (dash is represented with some code).
> 
> Fixed.

> > In tables Table I-1 - Table I-V, text is not vertically centered in DOC, unlike in PDF.
> 
> Fixed.

Except from those issues, following differences are noticed:

- Cross-reference isn't generated in the figure's title. Moreover, figures are numbered differently, but that it reported in https://github.com/metanorma/mn-samples-itu/issues/85.

PDF:
![image](https://user-images.githubusercontent.com/55683568/80690090-9c654500-8ace-11ea-861c-a2d9becb7fab.png)

Original:
![image](https://user-images.githubusercontent.com/55683568/80690110-a38c5300-8ace-11ea-88b6-3f1e14ccbcc1.png)

- Table I.5 is broken in PDF:

PDF:
![image](https://user-images.githubusercontent.com/55683568/80691758-fa932780-8ad0-11ea-8d1d-3967f55cf91b.png)

Original:
![image](https://user-images.githubusercontent.com/55683568/80691785-0383f900-8ad1-11ea-8f65-961d7fd5ff57.png)

- Heading repetition for the tables which span to multiple pages.

> I can't set document specifiic 'heading repetition' feature. Other DOCs don't have heading repetition. We have to decide common rule for such cases.

So far, heading repetition was noticed in this document and ITU H.782. I think that it is always used in DOCs generation when it comes to ITU recommendations. @ronaldtse, do we have a rule in this case?

Issues which were already noticed in previous documents, i.e. general issues:

- Scope is not generated well in PDF (includes additional Definitions subsection). -> will be reported as XML issue.

- ITU references include the word "Recommendation" in bibliography of the original document, unlike in PDF.

**Comment by Intelligent2013** [comment]
> In #60 (comment), it is written that some fixes were made for this document, but the issues are unfortunately still present in PDF uploaded 2 days ago. Could you please try generating PDF again?

PDF here:
https://github.com/metanorma/mn-native-pdf/blob/gh-pages/documents/itu-Y.3507-201812.pdf

> Title is not rendering properly in PDF (dash is represented with some code).

Issue in XML:
```
<title language="en" format="text/plain" type="main">Cloud computing&amp;#8201;&amp;#8212;&amp;#8201;Functional requirements of physical machine</title>
```

> In tables Table I-1 - Table I-V, text is not vertically centered in DOC, unlike in PDF.

I didn't change it. It should looks as in DOC.



> Cross-reference isn't generated in the figure's title.

Fixed.

> Moreover, figures are numbered differently, but that it reported in metanorma/mn-samples-itu#85.

As I understand in XML should be some indication how to number figures, tables. etc. (see comment https://github.com/metanorma/mn-native-pdf/issues/96#issuecomment-619466561), but it there isn't yet.



> Table I.5 is broken in PDF:

Looks like this is Apache FOP bug. The previous version of document was rendered by using SourceSerifPro font , instead of Times New Roman (because Times New Roman font can't be downloaded in workflow), but when it rendered by Times New Roman font, then there isn't such issue. So I think this is a bug Apache FOP or SourceSerifPro font or both. I have no idea how to fix it.

Latest here (rendered by Times New Roman): https://github.com/metanorma/mn-native-pdf/blob/gh-pages/documents/itu-Y.3507-201812.pdf



> ITU references include the word "Recommendation" in bibliography of the original document, unlike in PDF.

In XML I can't find 'Recommendation' word in bibliography or an indication that I should place 'Recommendation' as fixed text.

**Comment by ronaldtse** [comment]
> So far, heading repetition was noticed in this document and ITU H.782. I think that it is always used in DOCs generation when it comes to ITU recommendations

@anermina yes we always need table heading repetition.

**Comment by Intelligent2013** [comment]
> @anermina yes we always need table heading repetition.

@ronaldtse Is this rule for ITU flavor only, or for all flavors?

**Comment by ronaldtse** [comment]
@Intelligent2013 I think in general we want to have table heading repetition if a table spans multiple pages?

**Comment by Intelligent2013** [comment]
@ronaldtse yes, it's useful. Just I remember when I've develop first iso xslt I've turned off specially this feature, because there isn't repeat header in source...

**Comment by ronaldtse** [comment]
@Intelligent2013 MN XML is not supposed to provide the repeat header, right? Placing the repeat header is the job of the rendering output?

**Comment by Intelligent2013** [comment]
@ronaldtse MN XML tables just has thead subelement and we can process it as repeat header or as usual table row. I'll change xslt for processing it as repeat header.

**Comment by Intelligent2013** [comment]
Repeat header added for all PDF.

**Comment by ronaldtse** [comment]
@anermina are we good here? Thanks!

**Comment by anermina** [comment]
> @anermina are we good here? Thanks!

Following issue is still active:

> > Table I.5 is broken in PDF:
> 
> Looks like this is Apache FOP bug. The previous version of document was rendered by using SourceSerifPro font , instead of Times New Roman (because Times New Roman font can't be downloaded in workflow), but when it rendered by Times New Roman font, then there isn't such issue. So I think this is a bug Apache FOP or SourceSerifPro font or both. I have no idea how to fix it.

Similar comments as in https://github.com/metanorma/mn-native-pdf/issues/153#issuecomment-625803512 apply here, as well. General issues mentioned in this comment will be solved for all documents at once, so I would just like to draw attention to "spacing after dashes" issue, which applies in table entries here:

PDF:
![image](https://user-images.githubusercontent.com/55683568/81427129-97516700-915a-11ea-9a67-63dea35050da.png)

Original:
![image](https://user-images.githubusercontent.com/55683568/81427136-9c161b00-915a-11ea-924e-491f6b2b04eb.png)

**Comment by Intelligent2013** [comment]
> I would just like to draw attention to "spacing after dashes" issue, which applies in table entries here:

Fixed.

**Comment by opoudjis** [comment]
@Intelligent2013 @ronaldtse FYI: I am unwatching this repo, I have to cut my watches down to get any work done at all. If you need to draw my attention to something, please at me, or preferably raise a new ticket in one of the metanorma repos (which you have been doing).

**Comment by anermina** [comment]
@Intelligent2013, now we don't have the word "clause" in cross-references.

PDF:
![image](https://user-images.githubusercontent.com/55683568/82585595-86f7ae00-9b96-11ea-8d22-c9db6b05cf8d.png)

Original:
![image](https://user-images.githubusercontent.com/55683568/82585623-9676f700-9b96-11ea-8f56-6978ceaaf81d.png)

Generated DOC:
![image](https://user-images.githubusercontent.com/55683568/82585570-7cd5af80-9b96-11ea-93af-c63305d4c7c0.png)

**Comment by Intelligent2013** [comment]
@anermina This problem was discussed in https://github.com/metanorma/mn-native-pdf/issues/180#issuecomment-629909187:
> It is not the correct behaviour in ITU, for instance:
>ITU G.650 clause 6.1.1.1:
>>   The mode field diameter is determined from the far-field intensity distribution F2(), according to the definition given in clause 3.4.1.4.

So I've removed 'clause' in the latest xslt for ITU.

**Comment by anermina** [comment]
@Intelligent2013 But doesn't https://github.com/metanorma/mn-native-pdf/issues/180#issuecomment-629909187 say that dropping "Clause" in front of subclauses shouldn't be done in ITU flavor?

**Comment by ronaldtse** [comment]
And the references should be "clause x.x.x" not "Clause x.x.x". @anermina can you help file this to metanorma-itu? Thanks.

**Comment by Intelligent2013** [comment]
Fixed. References changed to "clause x.x.x".

**Comment by anermina** [comment]
> And the references should be "clause x.x.x" not "Clause x.x.x". @anermina can you help file this to metanorma-itu? Thanks.

Filed in https://github.com/metanorma/metanorma-itu/issues/208.

@Intelligent2013 , abbreviations are indented in PDF. Please remove indentation before the abbreviation.

PDF:
![image](https://user-images.githubusercontent.com/55683568/82715102-e981a480-9c91-11ea-8f46-d6fd70ecc306.png)

Generated DOC:
![image](https://user-images.githubusercontent.com/55683568/82715089-db338880-9c91-11ea-85fa-b8ffb1d7e5e7.png)

Comments were quite long here, so let me sum up the remaining issues:
- We need a rule for title repetition for the tables breaking into multiple pages. Heading should be (and is) repeated in generated PDF, but should the title also be repeated? It is in the original ITU document. In case title repetition is needed, I would also need to open a ticket in metanorma-itu.
- Table I.5 is broken in PDF. -> This one is hard to solve because it's likely to be an Apache FOP bug.

**Comment by Intelligent2013** [comment]
> abbreviations (and definition lists in general) are indented in PDF. Please remove indentation before the abbreviation/definition list.

Fixed.

**Comment by anermina** [comment]
Last columns of tables I.2-I.4 from Annex I are left-aligned in PDF, although center alignment is specified in .adoc. Please check.

PDF:
![image](https://user-images.githubusercontent.com/55683568/83909223-b0a0ff80-a768-11ea-86fb-361ee3f97bd8.png)

Original:
![image](https://user-images.githubusercontent.com/55683568/83909260-c282a280-a768-11ea-9e27-f31e80725269.png)

**Comment by Intelligent2013** [comment]
> Last columns of tables I.2-I.4 from Annex I are left-aligned in PDF, although center alignment is specified in .adoc.

Fixed.
![изображение](https://user-images.githubusercontent.com/5559926/83968464-32f90300-a8d2-11ea-9efc-e63bed0e2bdc.png)
